assets/app/scripts/filters/resources.js

```diff
@@ -20,6 +29,33 @@ angular.module('openshiftConsole')
       return null;
     };
   })  
+  .filter('icon', function(annotationFilter) {
```

Comment:
as discussed earlier we probably want to have an overarching, "get the icon for this thing" which can handle generating either an image tag or an <i> tag with a class.  Maybe it needs to be a directive since its going to be generating HTML.